iroha-lib/examples/DomainAssetCreation.cpp

```diff
@@ -0,0 +1,251 @@
+                torii_port)
+            .send(tx);
+
+    printTransactionStatus(
+                peer_ip,
+                torii_port,
+                getTransactionHash(tx));
+}
+
+void printAccountAssets(const std::string& account_name, const std::string& key_path, const std::string& peer_ip, uint16_t torii_port)
+{
+    std::cout << LINE << ">" << __FUNCTION__ << LINE << std::endl;
+
+    const auto query_proto = generateGetAccountAssetsQuery(
+                account_name,
+                key_path);
+    QueryResponse response = iroha_lib::GrpcClient(
+                peer_ip,
+                torii_port)
+            .send(query_proto);
+    auto payload = query_proto.payload();
+    assert(payload.get_account_assets().account_id() == account_name);
+    assert(payload.has_get_account_assets());
+
+    if (response.has_error_response())
+    {
+        const auto errorResponse = response.error_response();
+        std::cerr << errorResponse.error_code() << ": " << errorResponse.message() << std::endl;
+        return;
+    }
+
+    assert(response.has_account_assets_response());
+    auto accountAssetsResponce = response.account_assets_response();
+    for (const auto& r : response.account_assets_response().account_assets())
+    {
+        std::cout << "\tasset:" << r.asset_id() << " " << r.balance() << std::endl;
+    }
+
+    std::cout << LINE << "<" << __FUNCTION__ << LINE << std::endl;
+}
+
+iroha::protocol::Query generateGetAccountAssetsQuery(
+        const std::string& account_name,
+        const std::string& key_path)
+{
+    return generateQueryBase(account_name, key_path)
+            .getAccountAssets(account_name)
+            .signAndAddSignature();
+}
+
+iroha_lib::Query generateQueryBase(
+        const std::string& account_name,
+        const std::string& key_path)
+{
+    static auto log_manager = std::make_shared<logger::LoggerManagerTree>(
+                logger::LoggerConfig{logger::LogLevel::kInfo,
+                                     logger::getDefaultLogPatterns()})->getChild("CLI");
+
+    const auto keypair = generateKeypair(
+                account_name,
+                key_path,
+                log_manager);
+    return iroha_lib::Query(keypair);
+}
+
+iroha::protocol::Query generateGetAccountTransactionsQuery(
+        const std::string& account_name,
+        const std::string& key_path)
+{
+    return generateQueryBase(account_name, key_path)
+            .getAccountTransactions(account_name)
+            .signAndAddSignature();
+}
+
+iroha::protocol::Query generateGetAccountQuery(
+        const std::string& account_name,
+        const std::string& key_path)
+{
+    return generateQueryBase(account_name, key_path)
+            .getAccount(account_name)
+            .signAndAddSignature();
+}
+
+void printAccount(const std::string& account_name, const std::string& key_path, const std::string& peer_ip, uint16_t torii_port)
+{
+    std::cout << LINE << ">" << __FUNCTION__ << LINE << std::endl;
```

Comment:
These forms of debug statements should be handled by a proper logging library. I don't mind the extra overhead, if it means code that focuses on the right things and encourages the right way to do things. As it stands `example`s are a "do as I say, and not as I do" kind of deal.

Reply:
Here is new version: `fmt::print("----------->{}-----------\n", __FUNCTION__);`